src/Hamiltonians/Sum.jl

```diff
@@ -0,0 +1,136 @@
+"""
+    HamiltonianSum(A::AbstractHamiltonian, B::AbstractHamiltonian; a::Number=1, b::Number=1, weight::Real=1)
+    +(A::AbstractHamiltonian, B::AbstractHamiltonian)
+
+The sum of two [`AbstractHamiltonian`](@ref)s with coefficients, `aA + bB`. The two
+Hamiltonians must act on the same address space. The ratio of random spawns from `A` and
+spawns from `B` is controlled by `weight`.
+"""
+struct HamiltonianSum{T, H1<:AbstractHamiltonian, H2<:AbstractHamiltonian} <: AbstractHamiltonian{T}
+    h1::H1
+    h2::H2
+    a::Number
+    b::Number
```

Comment:
```suggestion
    a::T
    b::T
```
Those should be converted to `T` before being stored so we don't have type instabilites.

Reply:
Also do this for the column and offdiagonals below.